File: 0001-two-sum/0001-two-sum.py
from itertools import combinations


class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        vals = {}
        for i in range(len(nums)):
            if nums[i] in vals:
                return [vals[nums[i]], i]
            else:
                vals[target - nums[i]] = i

File: leetcode/Easy/0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> vals = new HashMap<>();
        
        for (int i = 0; i < nums.length; i++) {
            if (vals.containsKey(nums[i])) {
                return new int[] {vals.get(nums[i]), i};
            } else {
                vals.put(target - nums[i], i);
            }
        }
        
        return new int[] {0, 0};
    }
}